bin/check-rabbitmq-node-usage.rb

```diff
@@ -0,0 +1,306 @@
+#!/usr/bin/env ruby -W0
+#  encoding: UTF-8
+#
+# RabbitMQ check node health plugin
+# ===
+#
+# DESCRIPTION:
+#   This plugin checks if RabbitMQ server node is in a running state.
+#
+#   The plugin is based on the RabbitMQ cluster node health plugin by Tim Smith
+#   Edited my Milos Dodic and Milos Buncic (2016/10/27)
+#
+# USAGE:
+#   "check-rabbitmq-node-usage.rb -w host --username user --password pass --type mem -m 50 -c 90"
+#
+# NOTES:
+#   Use of "type" is mandatory. The script will check only the specified type.
+#
+# PLATFORMS:
+#   Linux, Windows, BSD, Solaris
+#
+# DEPENDENCIES:
+#   RabbitMQ rabbitmq_management plugin
+#   gem: sensu-plugin
+#   gem: carrot-top
+#
+# LICENSE:
+# Copyright 2012 Abhijith G <abhi@runa.com> and Runa Inc.
+# Copyright 2014 Tim Smith <tim@cozy.co> and Cozy Services Ltd.
+# Copyright 2015 Edward McLain <ed@edmclain.com> and Daxko, LLC.
+#
+# Released under the same terms as Sensu (the MIT license); see LICENSE
+# for details.
+
+require 'sensu-plugin/check/cli'
+require 'carrot-top'
+
+# Main plugin class
+class CheckRabbitMQNodeUsage < Sensu::Plugin::Check::CLI
+  option :host,
+         description: 'RabbitMQ host',
+         short: '-w',
+         long: '--host HOST',
+         default: 'localhost'
+
+  option :username,
+         description: 'RabbitMQ username',
+         short: '-u',
+         long: '--username USERNAME',
+         default: 'guest'
+
+  option :password,
+         description: 'RabbitMQ password',
+         short: '-p',
+         long: '--password PASSWORD',
+         default: 'guest'
+
+  option :port,
+         description: 'RabbitMQ API port',
+         short: '-P',
+         long: '--port PORT',
+         default: '15672'
+
+  option :ssl,
+         description: 'Enable SSL for connection to the API',
+         long: '--ssl',
+         boolean: true,
+         default: false
+
+  option :type,
+         description: 'Resource type',
+         required: true,
+         long: '--type TYPE',
+         in: %w(mem socket fd proc disk)
+
+  option :memwarn,
```

Comment:
if you are specifying type is there some reason to not just have standard warnings and crits? Whats the value added here?